app/models/sitewide_setting.rb

```diff
@@ -12,4 +11,20 @@ def name
     key.humanize
   end
 
+  def self.set(key, on)
+    flag = find_by_key_or_create(key)
```

Comment:
A more Rails 4 way to do this is to use `find_or_create_by`:

```ruby
find_or_create_by(key: key)
```

The magic finders are being deprecated in favour of this key-based approach. Also maybe worth using the bang version so it breaks with tears if the setting cannot be created rather than silently failing to create the setting - `find_or_create_by!`